addons/web/static/tests/views/layout_tests.js

```diff
@@ -0,0 +1,137 @@
+/** @odoo-module **/
+
+import { makeWithSearch, setupControlPanelServiceRegistry } from "@web/../tests/search/helpers";
+import { dialogService } from "@web/core/dialog/dialog_service";
+import { registry } from "@web/core/registry";
+import { Layout } from "@web/views/layout";
+
+const { Component, tags } = owl;
+const { xml } = tags;
+const serviceRegistry = registry.category("services");
+
+let serverData;
+
+QUnit.module("Views", (hooks) => {
+    hooks.beforeEach(() => {
+        serverData = {
+            models: {
+                foo: {
+                    fields: {
+                        aaa: {
+                            type: "selection",
+                            selection: [
+                                ["a", "A"],
+                                ["b", "B"],
+                            ],
+                        },
+                    },
+                    records: [],
+                },
+            },
+            views: {
+                "foo,false,search": /* xml */ `
+                    <search>
+                        <searchpanel>
+                            <field name="aaa" />
+                        </searchpanel>
+                    </search>`,
+            },
+        };
+
+        setupControlPanelServiceRegistry();
+        serviceRegistry.add("dialog", dialogService);
+    });
+
+    QUnit.module("Layout");
+
+    QUnit.test("Simple rendering", async (assert) => {
+        assert.expect(5);
+
+        class ToyComponent extends Component {}
+
+        ToyComponent.template = xml`
+            <Layout type="'toy'" useSampleModel="true">
+                <t t-set-slot="control-panel-top-right">
+                    <div class="toy_search_bar" />
+                </t>
+                <div class="toy_content" />
+            </Layout>`;
+        ToyComponent.components = { Layout };
+
+        const root = await makeWithSearch({
+            serverData,
+            Component: ToyComponent,
+            resModel: "foo",
+            searchViewId: false,
+        });
+
+        assert.hasClass(root.el, "o_toy_view o_view_sample_data");
+        assert.containsOnce(root, ".o_control_panel .o_cp_top_right .toy_search_bar");
+        assert.containsOnce(root, ".o_component_with_search_panel .o_search_panel");
```

Comment:
there is a searchpanel by default? If so it's consistent with the CP, but I'm not sure it's what we want

Reply:
The search panel is displayed if all the following conditions are met:
- a `<searchpanel>` node is found in the search arch;
- the `display.searchPanel` prop has not explicitely been set to a falsy value;
- and either the `view.type` of the search model is included in the `view_types` attribute on the search panel node (default: list & kanban) OR there is no view type on the model, in which case the current component is considered an action and should always display its search panel unless explicitely told.

These toy component don't give any view information and are considered actions. I think we want a search panel in such cases.